### DISCUSSION TITLE
[Q] How do I limit the domains if I enable sso

### DISCUSSION THREAD

**Discussion by nwns** [inventree/InvenTree#2619] [Q&A] [answered]
If I enable SSO using google, it allows all google accounts to login, I would like to limit valid emails to only those that are part of my Google Workspace Organization.  This does not seem to be addressed by django-allauth, which is fair since allauth is Authentication not Authorization and this is a question of which accounts should be Authorized to access the instance.

I have searched the docs for limit accounts and limit sso and I have not found anything.  Has this been addressed somewhere or is that a feature that has not been implemented?

**Comment by matmair** [comment, marked as answer]
@nwns I would block this at the authentication level to be safe.
Instead of using the google provider try using [open id](https://django-allauth.readthedocs.io/en/latest/providers.html#openid) and add the [hosted domain](https://developers.google.com/identity/protocols/oauth2/openid-connect#hd-param) parameter.
Would much prefer to handle things like this in config and not add code for more checks. There are some many ways to get those wrong.

Also please use discussions for questions, issues are more for FRs or bugs - discussions (QA type) are better for our SEO and threading is much more readable there.

**Reply by matmair** [reply]
@nwns have you tried this solution? Did it work?

**Reply by blieb** [reply]
I have the same problem. Think the most of the SSO supported providers is not useable for this app. You don't want to fully open your app for all users of google. 

Would be nice if there was also SAML2 support for single sign on. Then it is connected to your workspace account and only users with your workspace can sign in. 
I believe you can't limit a google app for your workspace only.

**Reply by matmair** [reply]
@blieb google work gives you the option to only access accounts of that org with OAuth2,OIDC or SAML. You can also set the default group to have no rights.
Feel free to implement saml2 support, it is not as straight forward as it seems if you want to implement the whole spec.

**Reply by blieb** [reply]
Hmm I was looking for that but I can only find an option for saml when i create a app in the admin console

**Reply by matmair** [reply]
Maybe they have changed it. When I implemented the last big SSO changes it was offered. I do not have access to a Google work admin console any longer so I can not really check.
Please raise an FR if you want a separate feature implemented in InvenTree.

**Reply by nwns** [reply]
Does Inventree support OAuth2, I will need check the docs, because that would solve my problem, we already have an LOB app that uses OAuth2 for login that is restricted to just our org.

I don't know about OIDC.  SAML is at admin.google.com (Security > Authenication).

Oauth2 is configured in console.cloud.google.com, you need a separate project for each OAuth2 application you are configuring. You need to be logged into the console as a user with sufficient permissions for the org, but it otherwise works the same as if you were using the console to make a public application that use the google apis.

**Reply by matmair** [reply]
InvenTree supports a lot of flavours of OAuth - you check in the [providers](https://django-allauth.readthedocs.io/en/latest/providers.html).

**Reply by blieb** [reply]
Ah yes I did find it at the cloud console now. You can choose internal when you create an app in your organisation.

**Reply by matmair** [reply]
@blieb Great, if you have further issues please open a new issue/discussion as it is hard to track already answered discussions

**Reply by matmair** [reply]
@blieb due to another request I will touch this part of the codebase again so I will implement this to ease this for future users. See https://github.com/inventree/InvenTree/issues/4168

**Reply by blieb** [reply]
> @blieb due to another request I will touch this part of the codebase again so I will implement this to ease this for future users. See https://github.com/inventree/InvenTree/issues/4168

Ok nice I will keep an eye on it. 

The other issue that I had is that I have to enable self signup to allow new users to login via oauth. But I only want users to sign up via oauth not via the sign up form. Would be nice if I can also enable sign up via Single sign on only. 

Shall I make a FR for that? I can also take a look. Think that is not hard to implement. But I haven't touch this language in years 😆

**Reply by matmair** [reply]
@blieb this is the FR I will take a look for. Ref #4166

**Reply by blieb** [reply]
@matmair oh nice. Was looking for it yesterday but did not find something about it. But this one is new. For now I just blocked the sign up page in Nginx 😆
